### ISSUE TITLE
HHVet change leaves NULL in tlsa_Exit.HHVet

### ISSUE THREAD

**Issue by eanders** [abtassociates/LSASampleCode#647]
The recent change: https://github.com/HMIS/LSASampleCode/commit/2363f20b8a32cb2ea322408aad148d1797bdeb34 now leaves some NULL values in the HHVet column in tlsa_Exit.  When these are pushed to `lsa_Exit` in step 7.8, the insert fails because `lsa_Exit` doesn't allow NULL values for `HHVet`.

In at least one case we're seeing, `tlsa_HHID.ExitCohort` is `-2` and `tlsa_Enrolllment.ExitAge2` is outside of 18-65.

One solution we're pondering is simply to update any `tlsa_Exit.HHVet` to `0` if it is null after step 7.4.1.  

@MollyMcEvilley, does that seem like a reasonable solution, or are we potentially missing data for other columns because the join is limited for HHVet?

**Comment by eanders** [comment]
It appears this limit is actually blocking other columns as well: `HHDisability, HHFleeingDV, HoHRace, HoHEthnicity, HHParent` all remain blank if the chosen age isn't between 18 and 65.  

Here's our proposed solution which moves the check for age into the select instead of having it as a limit:
```
/*
	7.4 Population Identifiers for LSAExit
*/

	update ex
	set HHVet = vet.vet
		, HHDisability = (select max(case when disability.DisabilityStatus = 1 then 1 else 0 end)
			from tlsa_Enrollment disability
			where disability.HouseholdID = hh.HouseholdID)
		, HHFleeingDV = (select max(case when dv.DVStatus = 1 then 1 else 0 end)
			from tlsa_Enrollment dv
			where dv.HouseholdID = hh.HouseholdID)
		, HoHRace =  case
			when hoh.RaceNone in (8,9) then 98
			when hoh.AmIndAkNative + hoh.Asian + hoh.BlackAfAmerican +
				hoh.NativeHIOtherPacific + hoh.White > 1 then 6
			when hoh.White = 1 and hoh.Ethnicity = 1 then 1
			when hoh.White = 1 then 0
			when hoh.BlackAfAmerican = 1 then 2
			when hoh.Asian = 1 then 3
			when hoh.AmIndAkNative = 1 then 4
			when hoh.NativeHIOtherPacific = 1 then 5
			else 99 end
		, HoHEthnicity = case
			when hoh.Ethnicity in (8,9) then 98
			when hoh.Ethnicity in (0,1) then hoh.Ethnicity
			else 99 end
		, HHParent = (select max(case when parent.RelationshipToHoH = 2 then 1 else 0 end)
			from tlsa_Enrollment parent
			where parent.HouseholdID = hh.HouseholdID)
		, ex.Step = '7.4.1'
	from tlsa_Exit ex
	inner join tlsa_HHID hhid on hhid.HouseholdID = ex.QualifyingExitHHID
	inner join hmis_Client hoh on hoh.PersonalID = ex.HoHID
	inner join (
		select n.HouseholdID, n.PersonalID, n.EnrollmentID, hhid.ExitCohort
			, n.RelationshipToHoH
		from tlsa_Enrollment n
		inner join tlsa_HHID hhid on hhid.HouseholdID = n.HouseholdID) hh on hh.HouseholdID = ex.QualifyingExitHHID and hh.ExitCohort = ex.Cohort
	inner join (select n.HouseholdID, hhid.ExitCohort, max(
		case
		when c.VeteranStatus = 1 and hhid.ExitCohort = 0 and n.ActiveAge between 18 and 65
		  or c.VeteranStatus = 1 and hhid.ExitCohort = -1 and n.Exit1Age between 18 and 65
		  or c.VeteranStatus = 1 and hhid.ExitCohort = -2 and n.Exit2Age between 18 and 65
		then
      1
		else
		  0
		end
	) as vet
		from tlsa_Enrollment n
		inner join hmis_Client c on c.PersonalID = n.PersonalID
		inner join tlsa_HHID hhid on hhid.HouseholdID = n.HouseholdID
		-- where case hhid.ExitCohort -- FIXME: This change leaves HHVet NULL
		-- 	when 0 then n.ActiveAge
		-- 	when -1 then n.Exit1Age
		-- 	when -2 then n.Exit2Age end between 18 and 65
		group by n.HouseholdID, hhid.ExitCohort) vet on vet.HouseholdID = ex.QualifyingExitHHID and vet.ExitCohort = ex.Cohort
```

**Comment by jbloescw** [comment]
We are seeing this also and is causing errors on known good uploads. 
Is the Inner Join causing the issue?  It could be turned into a left join and then pop either a case when or a coalesce in the Set HHVet section up top.

**Comment by eanders** [comment]
The new where clause in the inner join was causing the issue.  I wanted to change as little as possible of the logic, and I the original fix was put in to catch the odd 17 year old vet from #645.  The where was limiting the join where I think the logic should have been to simply use different dates when calculating veteran status.

**Comment by 2008zjqc** [comment]
Good afternoon,
We echo on this issue as well. The new code updates does causing error at step 7.8 when trying to insert into lsa_exit. This is due to the above query fix at 7.4.1, the result are saved at 'VET'  and only those adults has value with inner join. Then those household with child only will have no values for 'HHdisability', 'HHParent' etc which then error out at 7.8 as those part doesn't allow enter NULL for relevant columns. 

Thanks.
Chen

**Comment by MollyMcEvilley** [comment]
My sincere apologies for the bug and the delay in fixing it...
